chore: add RocksDB and EventManager builder configs

glevco opened the pull request:
Acceptance Criteria:

- Make `HathorManager` receive a required `EventStorage`
  - This will be necessary on https://github.com/HathorNetwork/hathor-core/pull/555, to check if the event queue feature was enabled in the previous run
- `Builder` changes
  - Add `peer_id` to builder artifacts
  - Make memory be the default storage type
  - Add RockDB's `cache_capacity` and `with_index` configs
  - Add `enable_event_manager()` auxiliary method
- Update usages of `Builder`, (`CliBuilder` and `Simulator`) to use event related configs

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
         self.check_if_can_modify()
         self._storage_type = StorageType.ROCKSDB
         self._rocksdb_path = path
+        self._with_rocksdb_index = with_index
```
Isn't be better to standardize with the `_rocksdb_` prefix? For instance, this one would be `_rocksdb_with_index`.

glevco replied:
Agreed. Done in https://github.com/HathorNetwork/hathor-core/pull/568/commits/33a1b40d435bb62afe13891b4c418ab5667bb54e

msbrogli replied:
Thanks!

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
+
         raise NotImplementedError
 
+    def _get_event_storage(self) -> EventStorage:
```
We must be careful because it's not safe to call this method twice. I feel it is the same for many other methods in the builder.

glevco replied:
Changed as explained in https://github.com/HathorNetwork/hathor-core/pull/568#issuecomment-1526563055

msbrogli replied:
Thanks!

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
+    def with_event_manager(self, *, event_ws_factory: EventWebsocketFactory) -> 'Builder':
+        self.check_if_can_modify()
+
+        self._event_manager = EventManager(
```
Why are you building while the builder is being configured?

glevco replied:
Changed as explained in https://github.com/HathorNetwork/hathor-core/pull/568#issuecomment-1526563055

msbrogli replied:
Thanks!

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
+
+        raise NotImplementedError
+
+    def with_event_manager(self, *, event_ws_factory: EventWebsocketFactory) -> 'Builder':
```
Why `with_` instead of `set_`? From the code, we might rename it to `enable_event_manager(...)`. What do you think?

glevco replied:
I didn't use `set_` because this method doesn't receive an `EventManager`, it receives an `EventWebsocketFactory` that is used (and required) to build the `EventManager` later.

I agree and changed it to `enable_event_manager()` in https://github.com/HathorNetwork/hathor-core/pull/568/commits/c10b4428a83ee376a7b68ccae29aeabf8a99137e

msbrogli replied:
Great! Thanks!

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
+
+        raise NotImplementedError
+
+    def with_event_manager(self, *, event_ws_factory: EventWebsocketFactory) -> 'Builder':
```
Why aren't you passing the event manager to the `HathorManager` in the `build()` method?

glevco replied:
Changed as explained in https://github.com/HathorNetwork/hathor-core/pull/568#issuecomment-1526563055

msbrogli replied:
Thanks!

glevco commented:
I have moved some changes from https://github.com/HathorNetwork/hathor-core/pull/555 to here in https://github.com/HathorNetwork/hathor-core/pull/568/commits/f1d9c96e5cff430a28b68943ada2700e54ac9c3b, as they resolve most of @msbrogli's review discussions. I have updated the Acceptance Criteria accordingly

msbrogli commented on hathor/manager.py:
```diff
@@ -87,6 +88,7 @@ def __init__(self,
                  consensus_algorithm: ConsensusAlgorithm,
                  peer_id: PeerId,
                  tx_storage: TransactionStorage,
+                 event_storage: EventStorage,
```
Why would `HathorManager.__ini__(...)` receive an `EventStorage` object?

msbrogli replied:
Why don't you add methods to the EventManager to handle it? I mean, not sure we need to surface EventStorage to do this.

msbrogli replied:
Perfect. Let's do it in a separate PR so you can move forward with your other PRs.

glevco replied:
Done in https://github.com/HathorNetwork/hathor-core/pull/555

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -158,16 +164,19 @@ def build(self) -> BuildArtifacts:
             checkpoints=self._checkpoints,
             capabilities=self._capabilities,
             environment_info=get_environment_info(self._cmdline, peer_id.id),
+            event_manager=event_manager,
```
Are we always instantiating an event manager? I thought it could be disabled.

glevco replied:
No, it can be disabled. The `_get_or_create_event_manager()` method in the `Builder` may return `None` if the `EventManager` was not set with `set_event_manager()` neither enabled with `enable_event_manager()`. The `HathorManager` also receives an `Optional[EventManager]`. Should it be passed in the kwargs instead?

msbrogli replied:
I guess it is fine. Thanks!

msbrogli commented on hathor/builder/builder.py:
```diff
@@ -239,30 +248,84 @@ def _get_or_create_pubsub(self) -> PubSubManager:
+                event_ws_factory=self._event_ws_factory
+            )
+
+        return self._event_manager
```
Shouldn't we assert that `self._event_ws_factory` is None?

glevco replied:
We could, but I left it like this in the same vein as letting the user set `use_memory()` and then `use_rocksdb()`. An user could use `enable_event_manager()` and then `set_event_manager()`.

msbrogli replied:
Got it. Thanks!
